transport.go

```diff
@@ -43,37 +40,28 @@ func (t *transport) RoundTrip(ctx *fasthttp.RequestCtx) {
 			ctx.Error("", fasthttp.StatusServiceUnavailable)
 			return
 		}
-		t.mu.Unlock()
-		if bytes.HasSuffix(umaeEndpoint, ctx.Path()) {
-			ctx.Success("text/plain", t.getUMAE())
-		}
-
-		finished := t.waiter.finishedCount()
-		if finished > 0 && finished%t.window.size() == 0 {
-			ctx.Error("", fasthttp.StatusServiceUnavailable)
-			go t.checkHeapAndGC()
-			return
-		}
 		t.waiter.requestArrived()
+		t.mu.Unlock()
 	}
 	ctx.Request.Header.Del("Connection")
 	if err := t.client.Do(&ctx.Request, &ctx.Response); err != nil {
 		panic(fmt.Sprintf("Problem calling proxy:%q", err))
 	}
 	ctx.Response.Header.Del("Connection")
 	if !*disableGCI {
-		t.mu.Lock()
-		t.waiter.requestFinished()
-		t.mu.Unlock()
+		if t.waiter.requestFinished()%t.window.size() == 0 {
+			// Yes, it is possible to call checkHeapAndGC twice
```

Comment:
Where there is another checkHeapAndGC call?

Reply:
And why is it possible?

Reply:
> Where there is another checkHeapAndGC call?

There is no other. Both would be called here. The explanation follows bellow.

> And why is it possible? to be called twice?

haha I was afraid of that question :) .. Imagine that there is a burst of 100 requests and that triggers the check heap and GC. Remember that there is no guarantee about when the goroutine will be started. Now imagine there is another burst of 100 requests and the runtime have not yet scheduled the first heap check goroutine for execution. This will trigger the routine again and it is bad.